### ISSUE TITLE
Should warn about redundant `cmd || exit $?` when `set -e`

### ISSUE THREAD

**Issue by zapashcanon** [koalaman/shellcheck#1163]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

set -e

ls || exit $?

ls
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

Using `cmd || exit $ ?` is redundant when using `set -e`.

**Comment by paride** [comment]
In my opinions scripts should be written to behave well even without `set -e`, which is in fact often discouraged:

https://mywiki.wooledge.org/BashFAQ/105

so I think "defensive" `|| exit` are fine.

**Comment by zapashcanon** [comment]
Actually, [Debian policy](https://www.debian.org/doc/debian-policy/ch-maintainerscripts.html) states that:

> For shell scripts this means that you almost always need to use set -e (this is usually true when writing shell scripts, in fact).

When you don't use `set -e`, shellcheck raise a warning about missing ` || exit`. But having to write this kind of things all the time is laborious. Using `set -e` makes it easier. But keeping `|| exit` and similar things is useless with `set -e` and also makes the script harder to read... This issue is only in case you use `set -e` (and clearly, there's only disadvantages with adding `|| exit` in this case), not about `set -e` should be used or not.
